### PR TITLE
Simplify: issue #10429 fixed

### PR DESCRIPTION
**Pull request by sandeepveethu**
Resolved issue #10429 . Added method for rewriting asinh as log function.

### CODE REVIEW

File: sympy/functions/elementary/hyperbolic.py [hunk 1]
```diff
@@ -846,6 +846,12 @@ def _eval_as_leading_term(self, x):
         else:
             return self.func(arg)
 
+    def _eval_rewrite_as_log(self,x):
+	    """
+	    Rewrites asinh as log function.
+	    """
+	    return log(x+sqrt(x**2+1))
```
Comment (leosartaj) [inline sympy/functions/elementary/hyperbolic.py]: ``log(x+sqrt(x**2+1))`` -> ``log(x + sqrt(x**2 + 1))``

File: sympy/functions/elementary/tests/test_hyperbolic.py [hunk 2]
```diff
@@ -439,6 +439,11 @@ def test_asinh():
     assert asinh(-I*(sqrt(5) + 1)/4) == -3*pi*I/10
 
 
+def test_asinh_rewrite():
+    x = Symbol('x')
+    assert asinh(x).rewrite(log) == log(x+sqrt(x**2+1))
```
Comment (leosartaj) [inline sympy/functions/elementary/tests/test_hyperbolic.py]: ``log(x+sqrt(x**2+1))`` -> ``log(x + sqrt(x**2 + 1))``

**Comment by leosartaj** [comment]
You need to fix code quality issues. Do not use tabs `always` use spaces. Other than this and my one comment this looks good.

**Comment by sandeepveethu** [comment]
@leosartaj ok, thanks a lot

File: sympy/functions/elementary/hyperbolic.py [hunk 3]
```diff
@@ -846,6 +846,12 @@ def _eval_as_leading_term(self, x):
         else:
             return self.func(arg)
 
+    def _eval_rewrite_as_log(self,x):
```
Comment (aktech) [inline sympy/functions/elementary/hyperbolic.py]: put a space after comma: `(self,x):` --> `(self, x):`

**Comment by sandeepveethu** [comment]
It got all messy for me , so I closed this PR and started afresh in PR #10439

**Pull request closed without merging**